terraform/main.tf

```diff
@@ -1,70 +1,135 @@
-  ecdsa_curve = "P384"
-}
+## -----------------------------------------------------------------------------------------
+# Define references to already existing Resource Group resources
+## -----------------------------------------------------------------------------------------
 
-locals {
-  dns_names = concat([for service in var.service_names : [service, "${service}.${var.namespace}", "${service}.${var.namespace}.svc"]]...)
+data "ibm_resource_group" "resource_group_cloudsvc" {
+  name = var.vpc_rg_cloudsvc
 }
 
-# Construct the CSR
-resource "tls_cert_request" "cert_request" {
-  for_each = toset(var.service_names)
-
-  key_algorithm   = "ECDSA"
-  private_key_pem = tls_private_key.private_key.private_key_pem
-  dns_names       = [each.key, "${each.key}.${var.namespace}", "${each.key}.${var.namespace}.svc"]
+data "ibm_resource_group" "resource_group_kube" {
+  name = var.vpc_rg_kube
+}
 
-  subject {
-    common_name  = each.key
-    organization = var.organization
-  }
+data "ibm_container_cluster_config" "cluster" {
+  cluster_name_id = var.cluster_name
+  resource_group_id = data.ibm_resource_group.resource_group_kube.id
+  admin	= true
 }
 
-# Issue the certificate signing request
-resource "kubernetes_certificate_signing_request" "csr" {
-  for_each = toset(var.service_names)
+## -----------------------------------------------------------------------------------------
+# Define reference to already existing namespaces
+## -----------------------------------------------------------------------------------------
+data "kubernetes_namespace" "spark_k8s_namespace" {
+  count = var.spark_kubernetes_namespace != "" ? 1 : 0
 
   metadata {
-    generate_name = "${each.key}-csr"
+    name = var.spark_kubernetes_namespace
   }
+}
 
-  spec {
-    usages  = ["client auth", "server auth"]
-    request = tls_cert_request.cert_request[each.key].cert_request_pem
-  }
-  auto_approve = true
+##############################################################################
+# Kubernetes Provider required to provision k8s resources
+##############################################################################
+
+provider kubernetes {
+  host                   = data.ibm_container_cluster_config.cluster.host
+  client_certificate     = data.ibm_container_cluster_config.cluster.admin_certificate
+  client_key             = data.ibm_container_cluster_config.cluster.admin_key
+  cluster_ca_certificate = data.ibm_container_cluster_config.cluster.ca_certificate
 }
 
-data "kubernetes_namespace" "namespace" {
-  metadata {
-    name = var.namespace
-  }
+##############################################################################
+# Kubernetes signed certificate creation for REST API server
+##############################################################################
+
+module "tls_cert" {
+  source                    = "./modules/tls_cert"
+  namespace                 = var.tsl_cert_namespace
+  service_names             = var.tsl_cert_service_names
```

Comment:
Are these variables suppose to de-reference var.tls_* or var.tsl_*?  ie. Is that a typo ?

Reply:
Good catch, fixed.